optuna/terminator/improvement/evaluator.py

```diff
@@ -59,41 +68,55 @@ def get_preprocessing(self, add_random_inputs: bool = False) -> BasePreprocessin
             ),
             UnscaleLog(),
             ToMinimize(),
+            OneToHot(),
         ]
 
-        if add_random_inputs:
-            processes += [AddRandomInputs(self._min_lcb_n_additional_samples)]
-
-        processes += [OneToHot()]
-
         return PreprocessingPipeline(processes)
 
     def evaluate(
         self,
-        trials: List[FrozenTrial],
+        trials: list[FrozenTrial],
         study_direction: StudyDirection,
     ) -> float:
         search_space = IntersectionSearchSpace().calculate(trials)
         self._validate_input(trials, search_space)
 
         fit_trials = self.get_preprocessing().apply(trials, study_direction)
```

Comment:
```suggestion
        trials = self.get_preprocessing().apply(trials, study_direction)
```

How about renaming `fit_trials` to `trials`, since we no longer need to distinguish between `lcb_trials` and `fit_trials`?

There seems to be a bug related to this line actually. Below, in `x, bounds = _convert_trials_to_tensors(trials)`, `trials` are not preprocessed.